src/Adapter/NewProducts/NewProductsProductSearchProvider.php

```diff
@@ -62,21 +62,23 @@ public function __construct(
      * @param ProductSearchQuery $query
      * @param string $type
      *
-     * @return array
+     * @return array|int
      */
     private function getProductsOrCount(
         ProductSearchContext $context,
         ProductSearchQuery $query,
         $type = 'products'
     ) {
-        return Product::getNewProducts(
+        $result = Product::getNewProducts(
             $context->getIdLang(),
             $query->getPage(),
             $query->getResultsPerPage(),
             $type !== 'products',
             $query->getSortOrder()->toLegacyOrderBy(),
             $query->getSortOrder()->toLegacyOrderWay()
         );
+
+        return $type !== 'products' ? (int) $result : $result;
```

Comment:
The `$type !== 'products'` is already done above in the call.

Reply:
Done